src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,148 @@
+            System.out.println("I can't seem to understand what task you are referring to.\n"
+                    + "Please let me know in this format: " + stringCommand
+                    + " <number of task>\n" + Ui.LINE);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Hmm... I don't have a task numbered " + answer + "\n" + Ui.LINE);
+        }
+    }
+
+    private static void furtherUnderstandTask(String answer, TaskList taskList) {
+        try {
+            String[] answers = answer.split(" ", 2);
+            if (answers.length > 1) {
+                String type = answers[0].strip();
+                String task = answers[1].strip();
+                String[] partsOfTask = task.split("/");
+
+                // checking for event and deadline if the date is given or not
+                if (type.equals(Commands.TODO.text) || partsOfTask.length == 2) {
+                    if (type.equals(Commands.TODO.text)) {
+                        Command command = new AddCommand(type);
+                        command.execute(task, taskList);
+
+                    } else {
+                        String description = partsOfTask[0].strip();
+                        String date = partsOfTask[1].strip();
+
+                        // check if the date is given in correct format
+                        if (date.length() != 8) {
+                            throw new DataFormatException();
+                        }
+
+                        Command command = new AddCommand(type);
+                        command.execute(task, taskList);
+
+                    }
+                } else {
+                    String instruction = "<type of task> <description> / <deadline>";
+                    if (type.equals(Commands.EVENT.text)) {
+                        instruction = "<type of task> <description> / <date of event>";
+                    }
+                    throw new DukeGotNoArgumentsException(instruction);
+                }
+            } else {
+                // handles the case when no description
+                String instruction = "<type of task> <description>";
+                if (answers[0].equals(Commands.DEADLINE.text)) {
+                    instruction = "<type of task> <description> / <due date>";
+                } else if (answers[0].equals(Commands.EVENT.text)) {
+                    instruction = "<type of task> <description> / <date of event>";
+                }
+                throw new DukeGotNoArgumentsException(instruction);
+            }
+        } catch (DukeGotNoArgumentsException e) {
+            System.out.println(e.getMessage() + "\n" + Ui.LINE);
+        } catch (DataFormatException e) {
+            System.out.println("Please key in again with the date in the ddmmyyyy format." + "\n" + Ui.LINE);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println(answer);
+        }
+    }
+
+    enum Commands {
```

Comment:
I didn't know this way of using enum before, nice implementation! However, it may be better if you create a class for Command, according to OOP.

Reply:
Ahaha I do have a class for commands, but just wanted to enumerate the command strings within this class, but I shall see if theres a better way to merge both. Thanks for highlighting.